File: 프로그래머스/unrated/181830. 정사각형으로 만들기/정사각형으로 만들기.java
class Solution {
    public int[][] solution(int[][] arr) {
        int row = arr.length, col = arr[0].length;
        int n = row > col ? row : col;
        int [][] answer = new int[n][n];
        for (int i = 0; i < row; i++)
            System.arraycopy(arr[i], 0, answer[i], 0, col);
        return answer;
    }
}

File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.js
function solution(arr) {
    var row = arr.length;
    var col = arr[0].length;
    if (row < col)
        for (var i = 0; i < col - row; i++)
            arr.push(new Array(col).fill(0));
    else if (col < row)
        for (var i = 0; i < row; i++)
            arr[i].push(... new Array(row - col).fill(0));
    return arr;
}